stack.go

```diff
@@ -27,14 +27,30 @@ func (s *stack) push(element interface{}) {
 	*s = append(*s, element)
 }
 
-// Remove and return top element of stack. Return false if stack is empty.
+// Remove and return top element of stack. Return error if stack is empty.
 func (s *stack) pop() (interface{}, error) {
 	if s.isEmpty() {
-		return nil, &InvalidOperationError{"stack", "Pop"}
+		return nil, &InvalidOperationError{structName: "stack", methodName: "pop"}
```

Comment:
We should be able to pass some more information here. pop is invalid because the stack is empty.

Reply:
Done.